service/matching/matching_engine.go

```diff
@@ -920,8 +920,27 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 					if physInfo, ok := physicalInfoByBuildId[buildId][taskQueueType]; !ok {
 						physicalInfoByBuildId[buildId][taskQueueType] = vii.PhysicalTaskQueueInfo
 					} else {
+						var bInfo *taskqueuepb.BacklogInfo
+
+						// only report BacklogInformation if requested.
+						if req.GetReportBacklogInfo() {
```

Comment:
When `req.GetBacklogInfo()` is `false`, no calls are being made to fetch the backlog info for each partition. If we don't have this `if statement`, it will result in `nil` objects being dereferenced and we shall receive panics.